pyrobosim/pyrobosim/navigation/astar_grid.py

```diff
@@ -0,0 +1,329 @@
+            f"Generating occupancy grid at resolution {self.resolution} and inflation radius {self.inflation_radius}"
+        )
+        ts = time.time()
+        self.grid = occupancy_grid_from_world(
+            self.world,
+            resolution=self.resolution,
+            inflation_radius=self.inflation_radius,
+        )
+        print(f"Generated in : {time.time() - ts} seconds.")
+
+    def _set_actions(self):
+        """Generates the actions available"""
+        self.actions = {
+            "left": {"cost": 1, "action": (-1, 0)},
+            "right": {"cost": 1, "action": (1, 0)},
+            "up": {"cost": 1, "action": (0, 1)},
+            "down": {"cost": 1, "action": (0, -1)},
+            "left_up": {"cost": 1.414, "action": (-1, 1)},
+            "left_down": {"cost": 1.414, "action": (-1, -1)},
+            "right_up": {"cost": 1.414, "action": (1, 1)},
+            "right_down": {"cost": 1.414, "action": (1, -1)},
+        }
+        keys = list(self.actions.keys())
+        self.selected_actions = keys if self.diagonal_motion else keys[:4]
+        print(f"Selected actions : {self.selected_actions}")
+
+    def _set_heuristic(self):
+        """Sets the heuristic"""
+        euclidean = lambda p1, p2: math.sqrt(
+            (p1[0] - p2[0]) ** 2 + (p1[1] - p2[1]) ** 2
+        )
+        manhattan = lambda p1, p2: abs(p1[0] - p2[0]) + abs(p1[1] - p2[1])
+
+        if self.distance_metric == "Euclidean":
+            self._heuristic = euclidean
+        elif self.distance_metric == "Manhattan":
+            if not self.diagonal_motion:
+                warnings.warn("Manhattan over estimates without diagonal motion")
+            self._heuristic = manhattan
+        print(f"Selected heuristic: {self.distance_metric}")
+
+    def _reset(self):
+        """Resets the state of data used by the algorithm"""
+        self.candidates = PriorityQueue()
+        self.visited.clear()
+        self.goal = None
+        self.start = None
+        self.latest_path = Path()
+        self.num_nodes_expanded = 0
+
+    def _expand(self, node):
+        """
+        Generates and adds free neighbours to exploration candidates
+        :param node: The node to be expanded from
+        :type node: :class: Node
+        """
+        for action_name in self.selected_actions:
+            cost = self.actions[action_name]["cost"]
+            action = self.actions[action_name]["action"]
+            _x = node.x + action[0]
+            _y = node.y + action[1]
+            visited = (_x, _y) in self.visited
+            occupied = self.grid.is_occupied(_x, _y)
+            if not visited and not occupied:
+                _g = node.g + cost
+                _h = self._heuristic((_x, _y), self.goal)
+                new_node = Node(_x, _y, _g, _h, parent=node)
+                self.candidates.put(new_node)
+                self.num_nodes_expanded += 1
+
+    def _get_best_candidate(self):
+        """Return the candidate with best metric"""
```

Comment:
Missing return value in docstring.... which is true of all docstrings added in this PR, actually